sitemedia/js/controllers/annotation_controller.js

```diff
@@ -1,64 +1,68 @@
+import { Controller } from "@hotwired/stimulus";
+import { useIntersection } from "stimulus-use";
 import * as Annotorious from "@recogito/annotorious-openseadragon";
 import {
     TranscriptionEditor,
     AnnotationServerStorage,
 } from "annotorious-tahqiq";
 
-import IIIFControler from "./iiif_controller";
-
-export default class extends IIIFControler {
-    static targets = [
-        "image",
-        "imageContainer",
-        "imageHeader",
-        "osd",
-        "rotation",
-        "rotationLabel",
-        "zoomSlider",
-        "zoomSliderLabel",
-        "zoomToggle",
-    ];
+export default class extends Controller {
+    static targets = ["image", "imageContainer"];
 
     connect() {
-        // enable deep zoom, annotorious-tahqiq on load
-
-        // load configuration variables from django settings; must use DOM query due to json_script
-        const config = JSON.parse(
-            document.getElementById("annotation-config").textContent
-        );
+        // enable intersection behaviors (appear, disappear)
+        useIntersection(this);
+    }
+    appear() {
+        // enable deep zoom, annotorious-tahqiq on first appearance
+        // (appear may fire subsequently, but nothing should happen then)
+        if (
+            !this.imageContainerTarget.querySelector(".openseadragon-container")
+        ) {
+            // load configuration variables from django settings; must use DOM query due to json_script
+            const config = JSON.parse(
+                document.getElementById("annotation-config").textContent
+            );
 
-        // TODO: Better way of determining if we're on mobile?
-        const isMobile = window.innerWidth <= 900;
+            // TODO: Better way of determining if we're on mobile?
```

Comment:
not urgent / immediate, but the todo made me think: we could think about using a media query for this instead; that would probably be more semantic, and it looks like you can watch for changes if you want